File: src/extras-ktexteditor/libmovingrangefeedback.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");

/// https://api.kde.org/ktexteditor-movingrangefeedback.html
pub const ktexteditor__movingrangefeedback = struct {
    /// New constructs a new KTextEditor::MovingRangeFeedback object.
    ///
    ///
    pub fn New() QtC.KTextEditor__MovingRangeFeedback {
        return qtc.KTextEditor__MovingRangeFeedback_new();
    }

    /// [Qt documentation](https://api.kde.org/ktexteditor-movingrangefeedback.html#rangeEmpty)
    ///
    /// ``` self: QtC.KTextEditor__MovingRangeFeedback, range: QtC.KTextEditor__MovingRange ```
    pub fn RangeEmpty(self: ?*anyopaque, range: ?*anyopaque) void {
        qtc.KTextEditor__MovingRangeFeedback_RangeEmpty(@ptrCast(self), @ptrCast(range));
    }

    /// [Qt documentation](https://api.kde.org/ktexteditor-movingrangefeedback.html#rangeEmpty)
    ///
    /// Allows for overriding the related default method
    ///
    /// ``` self: QtC.KTextEditor__MovingRangeFeedback, callback: *const fn (self: QtC.KTextEditor__MovingRangeFeedback, range: QtC.KTextEditor__MovingRange) callconv(.c) void ```
    pub fn OnRangeEmpty(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.KTextEditor__MovingRangeFeedback_OnRangeEmpty(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// [Qt documentation](https://api.kde.org/ktexteditor-movingrangefeedback.html#rangeEmpty)
    ///
    /// Base class method implementation
    ///
    /// ``` self: QtC.KTextEditor__MovingRangeFeedback, range: QtC.KTextEditor__MovingRange ```
    pub fn QBaseRangeEmpty(self: ?*anyopaque, range: ?*anyopaque) void {
        qtc.KTextEditor__MovingRangeFeedback_QBaseRangeEmpty(@ptrCast(self), @ptrCast(range));
    }

    /// [Qt documentation](https://api.kde.org/ktexteditor-movingrangefeedback.html#rangeInvalid)
    ///
    /// ``` self: QtC.KTextEditor__MovingRangeFeedback, range: QtC.KTextEditor__MovingRange ```
    pub fn RangeInvalid(self: ?*anyopaque, range: ?*anyopaque) void {
        qtc.KTextEditor__MovingRangeFeedback_RangeInvalid(@ptrCast(self), @ptrCast(range));
    }

    /// [Qt documentation](https://api.kde.org/ktexteditor-movingrangefeedback.html#rangeInvalid)
    ///
    /// Allows for overriding the related default method
    ///
    /// ``` self: QtC.KTextEditor__MovingRangeFeedback, callback: *const fn (self: QtC.KTextEditor__MovingRangeFeedback, range: QtC.KTextEditor__MovingRange) callconv(.c) void ```
    pub fn OnRangeInvalid(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.KTextEditor__MovingRangeFeedback_OnRangeInvalid(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// [Qt documentation](https://api.kde.org/ktexteditor-movingrangefeedback.html#rangeInvalid)
    ///
    /// Base class method implementation
    ///
    /// ``` self: QtC.KTextEditor__MovingRangeFeedback, range: QtC.KTextEditor__MovingRange ```
    pub fn QBaseRangeInvalid(self: ?*anyopaque, range: ?*anyopaque) void {
        qtc.KTextEditor__MovingRangeFeedback_QBaseRangeInvalid(@ptrCast(self), @ptrCast(range));
    }

    /// [Qt documentation](https://api.kde.org/ktexteditor-movingrangefeedback.html#mouseEnteredRange)
    ///
    /// ``` self: QtC.KTextEditor__MovingRangeFeedback, range: QtC.KTextEditor__MovingRange, view: QtC.KTextEditor__View ```
    pub fn MouseEnteredRange(self: ?*anyopaque, range: ?*anyopaque, view: ?*anyopaque) void {
        qtc.KTextEditor__MovingRangeFeedback_MouseEnteredRange(@ptrCast(self), @ptrCast(range), @ptrCast(view));
    }

    /// [Qt documentation](https://api.kde.org/ktexteditor-movingrangefeedback.html#mouseEnteredRange)
    ///
    /// Allows for overriding the related default method
    ///
    /// ``` self: QtC.KTextEditor__MovingRangeFeedback, callback: *const fn (self: QtC.KTextEditor__MovingRangeFeedback, range: QtC.KTextEditor__MovingRange, view: QtC.KTextEditor__View) callconv(.c) void ```
    pub fn OnMouseEnteredRange(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.KTextEditor__MovingRangeFeedback_OnMouseEnteredRange(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// [Qt documentation](https://api.kde.org/ktexteditor-movingrangefeedback.html#mouseEnteredRange)
    ///
    /// Base class method implementation
    ///
    /// ``` self: QtC.KTextEditor__MovingRangeFeedback, range: QtC.KTextEditor__MovingRange, view: QtC.KTextEditor__View ```
    pub fn QBaseMouseEnteredRange(self: ?*anyopaque, range: ?*anyopaque, view: ?*anyopaque) void {
        qtc.KTextEditor__MovingRangeFeedback_QBaseMouseEnteredRange(@ptrCast(self), @ptrCast(range), @ptrCast(view));
    }

    /// [Qt documentation](https://api.kde.org/ktexteditor-movingrangefeedback.html#mouseExitedRange)
    ///
    /// ``` self: QtC.KTextEditor__MovingRangeFeedback, range: QtC.KTextEditor__MovingRange, view: QtC.KTextEditor__View ```
    pub fn MouseExitedRange(self: ?*anyopaque, range: ?*anyopaque, view: ?*anyopaque) void {
        qtc.KTextEditor__MovingRangeFeedback_MouseExitedRange(@ptrCast(self), @ptrCast(range), @ptrCast(view));
    }

    /// [Qt documentation](https://api.kde.org/ktexteditor-movingrangefeedback.html#mouseExitedRange)
    ///
    /// Allows for overriding the related default method
    ///
    /// ``` self: QtC.KTextEditor__MovingRangeFeedback, callback: *const fn (self: QtC.KTextEditor__MovingRangeFeedback, range: QtC.KTextEditor__MovingRange, view: QtC.KTextEditor__View) callconv(.c) void ```
    pub fn OnMouseExitedRange(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.KTextEditor__MovingRangeFeedback_OnMouseExitedRange(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// [Qt documentation](https://api.kde.org/ktexteditor-movingrangefeedback.html#mouseExitedRange)
    ///
    /// Base class method implementation
    ///
    /// ``` self: QtC.KTextEditor__MovingRangeFeedback, range: QtC.KTextEditor__MovingRange, view: QtC.KTextEditor__View ```
    pub fn QBaseMouseExitedRange(self: ?*anyopaque, range: ?*anyopaque, view: ?*anyopaque) void {
        qtc.KTextEditor__MovingRangeFeedback_QBaseMouseExitedRange(@ptrCast(self), @ptrCast(range), @ptrCast(view));
    }

    /// [Qt documentation](https://api.kde.org/ktexteditor-movingrangefeedback.html#caretEnteredRange)
    ///
    /// ``` self: QtC.KTextEditor__MovingRangeFeedback, range: QtC.KTextEditor__MovingRange, view: QtC.KTextEditor__View ```
    pub fn CaretEnteredRange(self: ?*anyopaque, range: ?*anyopaque, view: ?*anyopaque) void {
        qtc.KTextEditor__MovingRangeFeedback_CaretEnteredRange(@ptrCast(self), @ptrCast(range), @ptrCast(view));
    }

    /// [Qt documentation](https://api.kde.org/ktexteditor-movingrangefeedback.html#caretEnteredRange)
    ///
    /// Allows for overriding the related default method
    ///
    /// ``` self: QtC.KTextEditor__MovingRangeFeedback, callback: *const fn (self: QtC.KTextEditor__MovingRangeFeedback, range: QtC.KTextEditor__MovingRange, view: QtC.KTextEditor__View) callconv(.c) void ```
    pub fn OnCaretEnteredRange(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.KTextEditor__MovingRangeFeedback_OnCaretEnteredRange(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// [Qt documentation](https://api.kde.org/ktexteditor-movingrangefeedback.html#caretEnteredRange)
    ///
    /// Base class method implementation
    ///
    /// ``` self: QtC.KTextEditor__MovingRangeFeedback, range: QtC.KTextEditor__MovingRange, view: QtC.KTextEditor__View ```
    pub fn QBaseCaretEnteredRange(self: ?*anyopaque, range: ?*anyopaque, view: ?*anyopaque) void {
        qtc.KTextEditor__MovingRangeFeedback_QBaseCaretEnteredRange(@ptrCast(self), @ptrCast(range), @ptrCast(view));
    }

    /// [Qt documentation](https://api.kde.org/ktexteditor-movingrangefeedback.html#caretExitedRange)
    ///
    /// ``` self: QtC.KTextEditor__MovingRangeFeedback, range: QtC.KTextEditor__MovingRange, view: QtC.KTextEditor__View ```
    pub fn CaretExitedRange(self: ?*anyopaque, range: ?*anyopaque, view: ?*anyopaque) void {
        qtc.KTextEditor__MovingRangeFeedback_CaretExitedRange(@ptrCast(self), @ptrCast(range), @ptrCast(view));
    }

    /// [Qt documentation](https://api.kde.org/ktexteditor-movingrangefeedback.html#caretExitedRange)
    ///
    /// Allows for overriding the related default method
    ///
    /// ``` self: QtC.KTextEditor__MovingRangeFeedback, callback: *const fn (self: QtC.KTextEditor__MovingRangeFeedback, range: QtC.KTextEditor__MovingRange, view: QtC.KTextEditor__View) callconv(.c) void ```
    pub fn OnCaretExitedRange(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.KTextEditor__MovingRangeFeedback_OnCaretExitedRange(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// [Qt documentation](https://api.kde.org/ktexteditor-movingrangefeedback.html#caretExitedRange)
    ///
    /// Base class method implementation
    ///
    /// ``` self: QtC.KTextEditor__MovingRangeFeedback, range: QtC.KTextEditor__MovingRange, view: QtC.KTextEditor__View ```
    pub fn QBaseCaretExitedRange(self: ?*anyopaque, range: ?*anyopaque, view: ?*anyopaque) void {
        qtc.KTextEditor__MovingRangeFeedback_QBaseCaretExitedRange(@ptrCast(self), @ptrCast(range), @ptrCast(view));
    }

    /// Delete this object from C++ memory.
    ///
    /// ``` self: QtC.KTextEditor__MovingRangeFeedback ```
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.KTextEditor__MovingRangeFeedback_Delete(@ptrCast(self));
    }
};

File: src_original/extras-ktexteditor/libmovingrangefeedback.cpp
#include <KTextEditor/MovingRange>
#define WORKAROUND_INNER_CLASS_DEFINITION_KTextEditor__MovingRangeFeedback
#define WORKAROUND_INNER_CLASS_DEFINITION_KTextEditor__View
#include <movingrangefeedback.h>
#include "libmovingrangefeedback.h"
#include "libmovingrangefeedback.hxx"

KTextEditor__MovingRangeFeedback* KTextEditor__MovingRangeFeedback_new() {
    return new VirtualKTextEditorMovingRangeFeedback();
}

void KTextEditor__MovingRangeFeedback_RangeEmpty(KTextEditor__MovingRangeFeedback* self, KTextEditor__MovingRange* range) {
    auto* vktexteditor__movingrangefeedback = dynamic_cast<VirtualKTextEditorMovingRangeFeedback*>(self);
    if (vktexteditor__movingrangefeedback && vktexteditor__movingrangefeedback->isVirtualKTextEditorMovingRangeFeedback) {
        self->rangeEmpty(range);
    } else {
        ((VirtualKTextEditorMovingRangeFeedback*)self)->rangeEmpty(range);
    }
}

void KTextEditor__MovingRangeFeedback_RangeInvalid(KTextEditor__MovingRangeFeedback* self, KTextEditor__MovingRange* range) {
    auto* vktexteditor__movingrangefeedback = dynamic_cast<VirtualKTextEditorMovingRangeFeedback*>(self);
    if (vktexteditor__movingrangefeedback && vktexteditor__movingrangefeedback->isVirtualKTextEditorMovingRangeFeedback) {
        self->rangeInvalid(range);
    } else {
        ((VirtualKTextEditorMovingRangeFeedback*)self)->rangeInvalid(range);
    }
}

void KTextEditor__MovingRangeFeedback_MouseEnteredRange(KTextEditor__MovingRangeFeedback* self, KTextEditor__MovingRange* range, KTextEditor__View* view) {
    auto* vktexteditor__movingrangefeedback = dynamic_cast<VirtualKTextEditorMovingRangeFeedback*>(self);
    if (vktexteditor__movingrangefeedback && vktexteditor__movingrangefeedback->isVirtualKTextEditorMovingRangeFeedback) {
        self->mouseEnteredRange(range, view);
    } else {
        ((VirtualKTextEditorMovingRangeFeedback*)self)->mouseEnteredRange(range, view);
    }
}

void KTextEditor__MovingRangeFeedback_MouseExitedRange(KTextEditor__MovingRangeFeedback* self, KTextEditor__MovingRange* range, KTextEditor__View* view) {
    auto* vktexteditor__movingrangefeedback = dynamic_cast<VirtualKTextEditorMovingRangeFeedback*>(self);
    if (vktexteditor__movingrangefeedback && vktexteditor__movingrangefeedback->isVirtualKTextEditorMovingRangeFeedback) {
        self->mouseExitedRange(range, view);
    } else {
        ((VirtualKTextEditorMovingRangeFeedback*)self)->mouseExitedRange(range, view);
    }
}

void KTextEditor__MovingRangeFeedback_CaretEnteredRange(KTextEditor__MovingRangeFeedback* self, KTextEditor__MovingRange* range, KTextEditor__View* view) {
    auto* vktexteditor__movingrangefeedback = dynamic_cast<VirtualKTextEditorMovingRangeFeedback*>(self);
    if (vktexteditor__movingrangefeedback && vktexteditor__movingrangefeedback->isVirtualKTextEditorMovingRangeFeedback) {
        self->caretEnteredRange(range, view);
    } else {
        ((VirtualKTextEditorMovingRangeFeedback*)self)->caretEnteredRange(range, view);
    }
}

void KTextEditor__MovingRangeFeedback_CaretExitedRange(KTextEditor__MovingRangeFeedback* self, KTextEditor__MovingRange* range, KTextEditor__View* view) {
    auto* vktexteditor__movingrangefeedback = dynamic_cast<VirtualKTextEditorMovingRangeFeedback*>(self);
    if (vktexteditor__movingrangefeedback && vktexteditor__movingrangefeedback->isVirtualKTextEditorMovingRangeFeedback) {
        self->caretExitedRange(range, view);
    } else {
        ((VirtualKTextEditorMovingRangeFeedback*)self)->caretExitedRange(range, view);
    }
}

// Base class handler implementation
void KTextEditor__MovingRangeFeedback_QBaseRangeEmpty(KTextEditor__MovingRangeFeedback* self, KTextEditor__MovingRange* range) {
    auto* vktexteditormovingrangefeedback = dynamic_cast<VirtualKTextEditorMovingRangeFeedback*>(self);
    if (vktexteditormovingrangefeedback && vktexteditormovingrangefeedback->isVirtualKTextEditorMovingRangeFeedback) {
        vktexteditormovingrangefeedback->setKTextEditor__MovingRangeFeedback_RangeEmpty_IsBase(true);
        vktexteditormovingrangefeedback->rangeEmpty(range);
    } else {
        self->KTextEditor::MovingRangeFeedback::rangeEmpty(range);
    }
}

// Auxiliary method to allow providing re-implementation
void KTextEditor__MovingRangeFeedback_OnRangeEmpty(KTextEditor__MovingRangeFeedback* self, intptr_t slot) {
    auto* vktexteditormovingrangefeedback = dynamic_cast<VirtualKTextEditorMovingRangeFeedback*>(self);
    if (vktexteditormovingrangefeedback && vktexteditormovingrangefeedback->isVirtualKTextEditorMovingRangeFeedback) {
        vktexteditormovingrangefeedback->setKTextEditor__MovingRangeFeedback_RangeEmpty_Callback(reinterpret_cast<VirtualKTextEditorMovingRangeFeedback::KTextEditor__MovingRangeFeedback_RangeEmpty_Callback>(slot));
    }
}

// Base class handler implementation
void KTextEditor__MovingRangeFeedback_QBaseRangeInvalid(KTextEditor__MovingRangeFeedback* self, KTextEditor__MovingRange* range) {
    auto* vktexteditormovingrangefeedback = dynamic_cast<VirtualKTextEditorMovingRangeFeedback*>(self);
    if (vktexteditormovingrangefeedback && vktexteditormovingrangefeedback->isVirtualKTextEditorMovingRangeFeedback) {
        vktexteditormovingrangefeedback->setKTextEditor__MovingRangeFeedback_RangeInvalid_IsBase(true);
        vktexteditormovingrangefeedback->rangeInvalid(range);
    } else {
        self->KTextEditor::MovingRangeFeedback::rangeInvalid(range);
    }
}

// Auxiliary method to allow providing re-implementation
void KTextEditor__MovingRangeFeedback_OnRangeInvalid(KTextEditor__MovingRangeFeedback* self, intptr_t slot) {
    auto* vktexteditormovingrangefeedback = dynamic_cast<VirtualKTextEditorMovingRangeFeedback*>(self);
    if (vktexteditormovingrangefeedback && vktexteditormovingrangefeedback->isVirtualKTextEditorMovingRangeFeedback) {
        vktexteditormovingrangefeedback->setKTextEditor__MovingRangeFeedback_RangeInvalid_Callback(reinterpret_cast<VirtualKTextEditorMovingRangeFeedback::KTextEditor__MovingRangeFeedback_RangeInvalid_Callback>(slot));
    }
}

// Base class handler implementation
void KTextEditor__MovingRangeFeedback_QBaseMouseEnteredRange(KTextEditor__MovingRangeFeedback* self, KTextEditor__MovingRange* range, KTextEditor__View* view) {
    auto* vktexteditormovingrangefeedback = dynamic_cast<VirtualKTextEditorMovingRangeFeedback*>(self);
    if (vktexteditormovingrangefeedback && vktexteditormovingrangefeedback->isVirtualKTextEditorMovingRangeFeedback) {
        vktexteditormovingrangefeedback->setKTextEditor__MovingRangeFeedback_MouseEnteredRange_IsBase(true);
        vktexteditormovingrangefeedback->mouseEnteredRange(range, view);
    } else {
        self->KTextEditor::MovingRangeFeedback::mouseEnteredRange(range, view);
    }
}

// Auxiliary method to allow providing re-implementation
void KTextEditor__MovingRangeFeedback_OnMouseEnteredRange(KTextEditor__MovingRangeFeedback* self, intptr_t slot) {
    auto* vktexteditormovingrangefeedback = dynamic_cast<VirtualKTextEditorMovingRangeFeedback*>(self);
    if (vktexteditormovingrangefeedback && vktexteditormovingrangefeedback->isVirtualKTextEditorMovingRangeFeedback) {
        vktexteditormovingrangefeedback->setKTextEditor__MovingRangeFeedback_MouseEnteredRange_Callback(reinterpret_cast<VirtualKTextEditorMovingRangeFeedback::KTextEditor__MovingRangeFeedback_MouseEnteredRange_Callback>(slot));
    }
}

// Base class handler implementation
void KTextEditor__MovingRangeFeedback_QBaseMouseExitedRange(KTextEditor__MovingRangeFeedback* self, KTextEditor__MovingRange* range, KTextEditor__View* view) {
    auto* vktexteditormovingrangefeedback = dynamic_cast<VirtualKTextEditorMovingRangeFeedback*>(self);
    if (vktexteditormovingrangefeedback && vktexteditormovingrangefeedback->isVirtualKTextEditorMovingRangeFeedback) {
        vktexteditormovingrangefeedback->setKTextEditor__MovingRangeFeedback_MouseExitedRange_IsBase(true);
        vktexteditormovingrangefeedback->mouseExitedRange(range, view);
    } else {
        self->KTextEditor::MovingRangeFeedback::mouseExitedRange(range, view);
    }
}

// Auxiliary method to allow providing re-implementation
void KTextEditor__MovingRangeFeedback_OnMouseExitedRange(KTextEditor__MovingRangeFeedback* self, intptr_t slot) {
    auto* vktexteditormovingrangefeedback = dynamic_cast<VirtualKTextEditorMovingRangeFeedback*>(self);
    if (vktexteditormovingrangefeedback && vktexteditormovingrangefeedback->isVirtualKTextEditorMovingRangeFeedback) {
        vktexteditormovingrangefeedback->setKTextEditor__MovingRangeFeedback_MouseExitedRange_Callback(reinterpret_cast<VirtualKTextEditorMovingRangeFeedback::KTextEditor__MovingRangeFeedback_MouseExitedRange_Callback>(slot));
    }
}

// Base class handler implementation
void KTextEditor__MovingRangeFeedback_QBaseCaretEnteredRange(KTextEditor__MovingRangeFeedback* self, KTextEditor__MovingRange* range, KTextEditor__View* view) {
    auto* vktexteditormovingrangefeedback = dynamic_cast<VirtualKTextEditorMovingRangeFeedback*>(self);
    if (vktexteditormovingrangefeedback && vktexteditormovingrangefeedback->isVirtualKTextEditorMovingRangeFeedback) {
        vktexteditormovingrangefeedback->setKTextEditor__MovingRangeFeedback_CaretEnteredRange_IsBase(true);
        vktexteditormovingrangefeedback->caretEnteredRange(range, view);
    } else {
        self->KTextEditor::MovingRangeFeedback::caretEnteredRange(range, view);
    }
}

// Auxiliary method to allow providing re-implementation
void KTextEditor__MovingRangeFeedback_OnCaretEnteredRange(KTextEditor__MovingRangeFeedback* self, intptr_t slot) {
    auto* vktexteditormovingrangefeedback = dynamic_cast<VirtualKTextEditorMovingRangeFeedback*>(self);
    if (vktexteditormovingrangefeedback && vktexteditormovingrangefeedback->isVirtualKTextEditorMovingRangeFeedback) {
        vktexteditormovingrangefeedback->setKTextEditor__MovingRangeFeedback_CaretEnteredRange_Callback(reinterpret_cast<VirtualKTextEditorMovingRangeFeedback::KTextEditor__MovingRangeFeedback_CaretEnteredRange_Callback>(slot));
    }
}

// Base class handler implementation
void KTextEditor__MovingRangeFeedback_QBaseCaretExitedRange(KTextEditor__MovingRangeFeedback* self, KTextEditor__MovingRange* range, KTextEditor__View* view) {
    auto* vktexteditormovingrangefeedback = dynamic_cast<VirtualKTextEditorMovingRangeFeedback*>(self);
    if (vktexteditormovingrangefeedback && vktexteditormovingrangefeedback->isVirtualKTextEditorMovingRangeFeedback) {
        vktexteditormovingrangefeedback->setKTextEditor__MovingRangeFeedback_CaretExitedRange_IsBase(true);
        vktexteditormovingrangefeedback->caretExitedRange(range, view);
    } else {
        self->KTextEditor::MovingRangeFeedback::caretExitedRange(range, view);
    }
}

// Auxiliary method to allow providing re-implementation
void KTextEditor__MovingRangeFeedback_OnCaretExitedRange(KTextEditor__MovingRangeFeedback* self, intptr_t slot) {
    auto* vktexteditormovingrangefeedback = dynamic_cast<VirtualKTextEditorMovingRangeFeedback*>(self);
    if (vktexteditormovingrangefeedback && vktexteditormovingrangefeedback->isVirtualKTextEditorMovingRangeFeedback) {
        vktexteditormovingrangefeedback->setKTextEditor__MovingRangeFeedback_CaretExitedRange_Callback(reinterpret_cast<VirtualKTextEditorMovingRangeFeedback::KTextEditor__MovingRangeFeedback_CaretExitedRange_Callback>(slot));
    }
}

void KTextEditor__MovingRangeFeedback_Delete(KTextEditor__MovingRangeFeedback* self) {
    delete self;
}
